guides/source/active_record_migrations.md

```diff
@@ -1224,15 +1530,25 @@ destroyed. Like anything which operates at the application level, these cannot
 guarantee referential integrity and so some people augment them with [foreign
 key constraints][] in the database.
 
-Although Active Record does not provide all the tools for working directly with
-such features, the `execute` method can be used to execute arbitrary SQL.
+In practice, foreign key constraints and unique indexes are generally considered
```

Comment:
The way I've phrased it is database level constraints add "stronger consistency guarantees", since they're enforced on the server versus the client Rails app.